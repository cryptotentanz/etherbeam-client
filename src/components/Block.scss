@use '../styles/resources/sizes';

.my-block {
  margin: sizes.$spacer 0px;
}
