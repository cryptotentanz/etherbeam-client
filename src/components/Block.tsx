import React, { FC, ReactNode } from 'react'
import './Block.scss'

interface BlockProps {
  className?: string
  children?: ReactNode
}

const Block: FC<BlockProps> = ({ className, children }: BlockProps) => {
  return <div className={['my-block', className].join(' ')}>{children}</div>
}

export default Block
