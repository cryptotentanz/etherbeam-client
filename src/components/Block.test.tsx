import React from 'react'
import { shallow, ShallowWrapper } from 'enzyme'
import Block from './Block'

describe('<Block />', () => {
  const content = 'Content.'
  let component: ShallowWrapper

  describe('Default', () => {
    beforeAll(() => {
      component = shallow(<Block>{content}</Block>)
    })

    it('should render', () => {
      const div = component.find('div').at(0)
      expect(div.hasClass('my-block')).toBeTruthy()
      expect(div.prop('children')).toBe(content)
    })

    it('should match snapshot', () => expect(component.render()).toMatchSnapshot())
  })

  describe('Class', () => {
    const className = 'my-class'

    beforeAll(() => {
      component = shallow(<Block className={className}>{content}</Block>)
    })

    it('should render', () => {
      const div = component.find('div').at(0)
      expect(div.hasClass('my-block')).toBeTruthy()
      expect(div.hasClass(className)).toBeTruthy()
    })
  })
})
