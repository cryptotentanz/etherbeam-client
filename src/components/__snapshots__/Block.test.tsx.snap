// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Block /> Default should match snapshot 1`] = `
<div
  class="my-block "
>
  Content.
</div>
`;
